### PR TITLE
feat: rename data types

### PR DESCRIPTION
**Pull request by mihai-sysbio**
**Related issue(s) and PR(s)**  
This PR, in combination with https://github.com/MetabolicAtlas/data-files/pull/33 and https://github.com/MetabolicAtlas/data-generation/pull/33, resolves https://github.com/MetabolicAtlas/MetabolicAtlas/issues/1179.

**Type of change**  

- [x] New feature (non-breaking change which adds functionality)
 
**List of changes made**  

- The text that is shown in the data types dropdown of the data overlay has been updated to a more accurate description.
- `README.md` documents a new command that was made possible in the switch to Vue 3.
- Some dangling code after the IP redo has been removed

**Screenshot of the fix**  

**Testing**  

- A certain branch in other repos is needed: https://github.com/MetabolicAtlas/data-files/pull/33 and https://github.com/MetabolicAtlas/data-generation/pull/33
- A rebuild is needed due to new dependencies
- Relative urls that can be reused both for production and local testing
- Instructions on how to test: see that the data overlay functionality remains the same in the _Map Viewer_ and _Interaction Partners_, and that entry by url also works, e.g., [/map-viewer/androgen_metabolism](http://localhost/explore/Human-GEM/map-viewer/androgen_metabolism?dim=2d&panel=1&sel=&search=&coords=-3404.62,-2584.39,0.38,0,0,500&dataTypes=reaction,gene&dataSources=HPA_single-cell_reactions.tsv,hpaRna.tsv&dataSets=adipose%20tissue_macrophages_17,None)

**Further comments**  

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [ ] Tests and lint/format validations are passing
- [ ] My changes generate no new warnings

### CODE REVIEW

File: frontend/src/components/explorer/mapViewer/DataOverlay.vue [hunk 1]
```diff
@@ -240,6 +241,7 @@ export default {
           type,
           propagate: false,
           index,
+          ...DATA_TYPES_COMPONENTS[type],
```
Comment (mihai-sysbio) [inline frontend/src/components/explorer/mapViewer/DataOverlay.vue:244]: Sorry, I didn't see this line was added in the commit:
```suggestion
          ...DATA_TYPES_COMPONENTS[type],
```

File: frontend/src/helpers/dataOverlay.js [hunk 2]
```diff
@@ -3,17 +3,20 @@ const DEFAULT_METABOLITE_COLOR = '#9df';
 const DEFAULT_REACTION_COLOR = '#fff';
 
 const DATA_TYPES_COMPONENTS = {
-  transcriptomics: {
+  gene: {
+    description: 'gene (e.g., transcriptomics, proteomics, etc.)',
```
Comment (nanjiangshu) [inline frontend/src/helpers/dataOverlay.js:7]: If we follow British English, `e.g.` is usually not followed by comma, i.e. 

`e.g. transcriptomics, proteomics, etc.)`

File: frontend/src/helpers/dataOverlay.js [hunk 3]
```diff
@@ -3,17 +3,20 @@ const DEFAULT_METABOLITE_COLOR = '#9df';
...
-  metabolomics: {
+  metabolite: {
+    description: 'metabolite (e.g., metabolomics, etc.)',
     componentType: 'metabolite',
     className: 'met',
     defaultColor: DEFAULT_METABOLITE_COLOR,
   },
-  fluxomics: {
+  reaction: {
+    description: 'reaction (e.g., reaction presence, fluxomics, kcat, etc.)',
```
Comment (nanjiangshu) [inline frontend/src/helpers/dataOverlay.js:19]: I'm not sure if it is the best way to follow the data type with examples. 
Maybe we can show the data type as just `gene` or `metabolite` while add a `info` icon to provide details.

**Review by nanjiangshu — COMMENTED** [review]
Nice work @mihai-sysbio, it works very well when I tested. Just left some minor comments regarding the description field for data type names.

File: frontend/src/helpers/dataOverlay.js (hunk 3 above)
Comment (mihai-sysbio) [reply frontend/src/helpers/dataOverlay.js:19]: NIce, this gave me an idea for a new commit, so I'm marking the PR as draft until then.

File: frontend/src/components/explorer/mapViewer/DataOverlay.vue [hunk 4]
```diff
@@ -211,24 +211,25 @@ export default {
...
       }
       return this.dataSourcesIndex;
     },
     filteredDataTypes() {
-      // Do not show fluxomics data for the interaction partners page
+      // Do not show reactionType data for the interaction partners page
       // The data type may still be selected, but not shown
       if (this.$route.name === 'interaction-details') {
-        const dataTypes = this.dataTypes.filter(elem => elem.name !== 'fluxomics');
+        const dataTypes = this.dataTypes.filter(elem => elem.name !== 'reactionType');
```
Comment (e0) [inline frontend/src/components/explorer/mapViewer/DataOverlay.vue:224]: The IP page is currently broken (`reaction` still shows up under data type select). It seems like `reactionType` should just be `reaction` here.

```suggestion
        // do not include reaction data for the interaction partners page
        const { reaction, ...dataSourcesIndex } = this.dataSourcesIndex;
        return dataSourcesIndex;
      }
      return this.dataSourcesIndex;
    },
    filteredDataTypes() {
      // Do not show reaction data for the interaction partners page
      // The data type may still be selected, but not shown
      if (this.$route.name === 'interaction-details') {
        const dataTypes = this.dataTypes.filter(elem => elem.name !== 'reaction');
```

File: frontend/src/components/explorer/mapViewer/DataOverlay.vue [hunk 5]
```diff
@@ -211,24 +211,25 @@ export default {
...
         return dataTypes;
       }
       return this.dataTypes;
     },
   },
   async created() {
     await this.getDataSourcesIndex(this.model.short_name);
+    this.DATA_TYPES_COMPONENTS = DATA_TYPES_COMPONENTS;
```
Comment (e0) [inline frontend/src/components/explorer/mapViewer/DataOverlay.vue:232]: The idiomatic Vue way of doing this is to add `DATA_TYPES_COMPONENT` to `data()`. In other words add `DATA_TYPES_COMPONENTS,` after `showModal: false,`.

**Review by e0 — CHANGES_REQUESTED** [review]

**Comment by inghylt** [comment]
I'm not a fan of the lengthy and to me confusing text of the data type drop down. I would prefer the text to just be `gene`, `reaction` etc. and have more information in the documentation if needed. I won't block merging this in but I won't approve as of now either

File: frontend/src/components/explorer/mapViewer/DataOverlay.vue (hunk 4 above)
Comment (mihai-sysbio) [reply frontend/src/components/explorer/mapViewer/DataOverlay.vue:227]: That's right, after implementing the change with `reactionType` I switched to `reaction` - and it seems I haven't re-tested everything. Thanks for catching this.

Comment (mihai-sysbio) [reply frontend/src/components/explorer/mapViewer/DataOverlay.vue]: This was implemented separately through commit [0c3a8fa](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1181/commits/0c3a8fa625ca66bbc821435b1e09e9a414363057). I had to force-push a couple of times till I got the syntax for `Co-authored-by` right.

File: frontend/src/helpers/dataOverlay.js (hunk 3 above)
Comment (mihai-sysbio) [reply frontend/src/helpers/dataOverlay.js]: Based on your and @inghylt's feedback, I have reimplented this by moving the data type description below the dropdown in commit [4a44ab2](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1181/commits/4a44ab256970a9a37ce7235905d3c3c5139d0c8f):

image

Based on this, I'll go ahead and request another review.

File: frontend/src/components/explorer/mapViewer/DataOverlay.vue (hunk 5 above)
Comment (mihai-sysbio) [reply frontend/src/components/explorer/mapViewer/DataOverlay.vue]: This was resolved through [4a44ab2](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1181/commits/4a44ab256970a9a37ce7235905d3c3c5139d0c8f).

**Comment by inghylt** [comment]
@mihai-sysbio I think that the changes introduced in 44ea1da created a bit weird UI of the cards. Talked to @MalinAhlberg about it and we came up with the changes introduced in 3d480ec. Note that we intentionally kept the changes introduced in the custom overlay modal the way they were as we thought they made sense there!

**Comment by mihai-sysbio** [comment]
Thanks for the commits @inghylt and @e0. I do see the similarity of the icon with the one above.
image

The difference though is that while one requires clicking (in blue), the new one requires holding the mouse in a small fixed position, and there is a delay until the text appears. Also, this is not going to work on touchscreens I guess.

The styling of the pop-over is also not under our control as of now, and the text looks fairly small on all browsers (about half of the current text).
image

> created a bit weird UI of the cards

I'm not sure I know what the weird part is, specifically. Maybe that it breaks the "flow"? If so, could that be sorted out with a different style font?


tl;dr: The rename of the data types has to go in as part of this PR, but the description is optional. So either it's merged in and improved in another issue, or the addition of the description is simply skipped in this PR.

**Comment by inghylt** [comment]
@mihai-sysbio I'm 100% in favor of just renaming and removing the description, IMO it is superfluous

**Comment by MalinAhlberg** [comment]
I also agree that the descriptions could be skipped (for now) in the overlay sidebar, but I think that the changes made in the modal for uploading data makes sense. Eg keeping "Add data for:" in the sidebar card, and no descriptions, and in the upload custom data modal, the text could be "These data are for" and the descriptions kept.

**Comment by inghylt** [comment]
image
image

The result of commit d34609ec82a85972ae3032c408906bb6398f0ff9, implementing the suggestion made in this comment https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1181#issuecomment-1298729125

**Review by inghylt — APPROVED** [review]

**Review by e0 — APPROVED** [review]
Quite some differing opinions on this issue but glad we can wrap it up for now.

**Review by nanjiangshu — APPROVED** [review]
I have the same opinion as both @inghylt and @MalinAhlberg, that the data type field just show the data type, i.e. `gene` or `reaction`, without description. If the info-icon solution does not look good we can just skip it. If we really want to add some explanation of the data type `gene` or reaction, it can be added in the `Documentation - Data overlay` section. 
I approve this PR with or without the added explanation in the Documentation.

File: frontend/src/helpers/dataOverlay.js (hunk 3 above)
Comment (mihai-sysbio) [reply frontend/src/helpers/dataOverlay.js]: This was resolved across several commits, and finally approved by @nanjiangshu.
